### PR TITLE
Inability to create a gateway when a schema is provided directly

### LINKED ISSUE #74: It is not possible to create a gateway when -s flag is used
Executing the gateway create command with the `-s` flag results in error.

Example: 

`quick gateway create -s schema.graphql schematest` (`quick gateway create --schema schema.graphql schematest`)

results in:
```
Internal Server Error: An unexpected error occurred:while parsing a block mapping
 in 'reader', line 1, column 1:
    apiVersion: v1
    ^
expected <block end>, but found '}'
 in 'reader', line 14, column 1:
    }
    ^
```

schema.graphql is:
```
type Query {
    getPerson(id: ID): Person @topic(name: "person", keyArgument: "id")
    getCorporate(id: ID): Corporate @topic(name: "corporate", keyArgument: "id")
}

type Person {
    id: ID!
    corporateId: ID!
    firstName: String
    lastName: String
    # birthday: String
    # birthLocation: String
    corporate: Corporate @topic(name: "corporate", keyField: "corporateId")
}

type Corporate {
    id: ID!
    referenceId: String
    name: String
    street: String
    city: String
}
```
There is a problem with transferring the schema.graphql to the .yml file with themeleaf. After having corrected the schema.graphql manually so that the indentation is 2 spaces and not 4, the error changes to:
```
mapping values are not allowed here
 in 'reader', line 12, column 35:
      findPurchase(purchaseId: String): Purchase @topic(name: "purchas ...
```

### PR DESCRIPTION
**Pull request by DawidNiezgodka**
Resolves #74.

### CODE REVIEW

File: manager/src/main/java/com/bakdata/quick/manager/gateway/KubernetesGatewayService.java [hunk 1]
```diff
@@ -85,8 +85,9 @@ public Single<GatewayDescription> getGateway(final String name) {
 
     @Override
     public Completable createGateway(final GatewayCreationData gatewayCreationData) {
-        return Single.fromCallable(() -> this.loader.forCreation(gatewayCreationData, ResourcePrefix.GATEWAY))
-            .flatMapCompletable(this.kubernetesManagerClient::deploy);
+            log.debug("Creating gateway without provided schema.");
```
Comment (torbsto) [inline manager/src/main/java/com/bakdata/quick/manager/gateway/KubernetesGatewayService.java:88]: Why without?

File: manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java [hunk 2]
```diff
@@ -205,7 +205,29 @@ void shouldCreateConfigMap() {
 
     @Test
     void shouldCreateConfigmapWithSchema() {
-        final String schema = "type Query { test: Int }";
+        final String schema = "type Query {\n" +
```
Comment (torbsto) [inline manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java:208]: Please move into a separate text file and load that file

File: manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java [hunk 3]
```diff
@@ -205,7 +205,29 @@ void shouldCreateConfigMap() {
...
+            "type Product {\n" +
+            "productId: Int!\n" +
+            "name: String\n" +
+            "description: String \n" +
+            "price: Price  }\n" +
+            "type Price {\n" +
+            "total: Float\n" +
+            "currency: String  }";
+
+        Mockito.when(this.gatewayClient.updateSchema(GATEWAY_NAME, new SchemaData(schema)))
```
Comment (torbsto) [inline manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java:229]: This isn't required anymore, is it?

File: manager/src/main/java/com/bakdata/quick/manager/gateway/resource/GatewayResourceLoader.java [hunk 4]
```diff
@@ -234,8 +237,14 @@ private ConfigMap createGatewayConfigMap(final String name, @Nullable final Stri
         final Context root = new Context();
         root.setVariable("name", name);
         if (schema != null) {
-            root.setVariable("schema", schema);
+            final String reformattedSchema = this.formatSchemaForYaml(schema);
+            log.debug("Writing the following schema: {}", reformattedSchema);
```
Comment (torbsto) [inline manager/src/main/java/com/bakdata/quick/manager/gateway/resource/GatewayResourceLoader.java:241]:
```suggestion
            log.debug("Create gateway config map with the following schema: {}", reformattedSchema);
```

File: manager/src/main/java/com/bakdata/quick/manager/gateway/KubernetesGatewayService.java [hunk 5]
```diff
@@ -97,6 +98,7 @@ public Completable deleteGateway(final String name) {
 
     @Override
     public Completable updateSchema(final String name, final String graphQLSchema) {
+        log.debug("Updating schema...");
```
Comment (torbsto) [inline manager/src/main/java/com/bakdata/quick/manager/gateway/KubernetesGatewayService.java:101]:
```suggestion
        log.debug("Updating schema of gateway {}", name);
```

File: manager/src/main/java/com/bakdata/quick/manager/gateway/resource/GatewayResourceLoader.java [hunk 6]
```diff
@@ -234,8 +237,14 @@ private ConfigMap createGatewayConfigMap(final String name, @Nullable final Stri
...
-            root.setVariable("schema", schema);
+            final String reformattedSchema = this.formatSchemaForYaml(schema);
+            log.debug("Creating gateway config map with the following schema: {}", reformattedSchema);
+            root.setVariable("schema", reformattedSchema);
         }
         return this.kubernetesResources.loadResource(root, "gateway/config-map", ConfigMap.class);
     }
+
+    private String formatSchemaForYaml(final String schema) {
+        return schema.lines().map(line -> "  " + line).collect(Collectors.joining());
```
Comment (raminqaf) [inline manager/src/main/java/com/bakdata/quick/manager/gateway/resource/GatewayResourceLoader.java:248]: Could you please put `"  "` in a const variable?

File: manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java [hunk 7]
```diff
@@ -204,9 +210,10 @@ void shouldCreateConfigMap() {
     }
 
     @Test
-    void shouldCreateConfigmapWithSchema() {
-        final String schema = "type Query { test: Int }";
-        this.createGateway(GATEWAY_NAME, 1, null, schema);
+    void shouldCreateConfigmapWithSchema(final TestInfo testInfo) throws IOException {
```
[the remark this answers is not in the document]
Comment (DawidNiezgodka) [reply manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java:213]: Isn't it tested in `KubernetesGatewayServiceTest.shouldCreateConfigMap`?

[the remark this answers is not in the document]
Comment (DawidNiezgodka) [reply manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java:213]: Or do you mean with an empty schema?

File: manager/src/main/resources/k8s/templates/gateway/config-map.th.yaml [hunk 8]
```diff
@@ -8,4 +8,4 @@ metadata:
     app.kubernetes.io/component: gateway
 data:
   schema.graphql: |
-    [(${schema})]
+  [(${schema})]
```
[the remark this answers is not in the document]
Comment (DawidNiezgodka) [reply manager/src/main/resources/k8s/templates/gateway/config-map.th.yaml:11]: Nope :)

File: manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (raminqaf) [reply manager/src/test/java/com/bakdata/quick/manager/gateway/KubernetesGatewayServiceTest.java:213]: Yes, with an empty schema. Az if we just call `quick gateway create bla-bla`

File: manager/src/main/java/com/bakdata/quick/manager/gateway/KubernetesGatewayService.java [hunk 9]
```diff
@@ -97,6 +97,7 @@ public Completable deleteGateway(final String name) {
 
     @Override
     public Completable updateSchema(final String name, final String graphQLSchema) {
+        log.debug("Updating schema of the gateway {}", name);
```
Comment (raminqaf) [inline manager/src/main/java/com/bakdata/quick/manager/gateway/KubernetesGatewayService.java:100]: Since you are at it. Add the schema as well

**Review by raminqaf — APPROVED** [review]

File: manager/src/test/resources/schema/graphql/shouldCreateConfigmapWithSchema.graphql [hunk 10]
```diff
@@ -0,0 +1,20 @@
+type Query {
```
Comment (torbsto) [inline manager/src/test/resources/schema/graphql/shouldCreateConfigmapWithSchema.graphql:1]: Is there still a reason why this is formatted like this?

**Review by torbsto — APPROVED** [review]

**Pull request merged**